internal/provider/resource_apply_module.go

```diff
@@ -0,0 +1,596 @@
+
+	return nil
+}
+
+// ApplyModuleModel is the model for an apply_module.
+// Please note: Unlike many/most other resources, this model does not exist in the Tharsis API.
+// The workspace path, module source, and module version uniquely identify this apply_module.
+type ApplyModuleModel struct {
+	WorkspacePath types.String        `tfsdk:"workspace_path"`
+	ModuleSource  types.String        `tfsdk:"module_source"`
+	ModuleVersion types.String        `tfsdk:"module_version"`
+	Variables     basetypes.ListValue `tfsdk:"variables"`
+}
+
+// Ensure provider defined types fully satisfy framework interfaces
+var (
+	_ resource.Resource                = (*applyModuleResource)(nil)
+	_ resource.ResourceWithConfigure   = (*applyModuleResource)(nil)
+	_ resource.ResourceWithImportState = (*applyModuleResource)(nil)
+)
+
+// NewApplyModuleResource is a helper function to simplify the provider implementation.
+func NewApplyModuleResource() resource.Resource {
+	return &applyModuleResource{}
+}
+
+type applyModuleResource struct {
+	client *tharsis.Client
+}
+
+// Metadata returns the full name of the resource, including prefix, underscore, instance name.
+func (t *applyModuleResource) Metadata(ctx context.Context,
+	req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "tharsis_apply_module"
+}
+
+func (t *applyModuleResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	description := "Defines and manages an apply module."
+
+	resp.Schema = schema.Schema{
+		Version:             1,
+		MarkdownDescription: description,
+		Description:         description,
+		Attributes: map[string]schema.Attribute{
+			"workspace_path": schema.StringAttribute{
+				MarkdownDescription: "The full path of the workspace.",
+				Description:         "The full path of the workspace.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"module_source": schema.StringAttribute{
+				MarkdownDescription: "The source of the module.",
+				Description:         "The source of the module.",
+				Required:            true,
+			},
+			"module_version": schema.StringAttribute{
+				MarkdownDescription: "The version identifier of the module.",
+				Description:         "The version identifier of the module.",
+				Optional:            true,
+				Computed:            true, // computed if not supplied
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"variables": schema.ListNestedAttribute{
+				MarkdownDescription: "Optional list of variables for the run in the target workspace.",
+				Description:         "Optional list of variables for the run in the target workspace.",
+				Optional:            true,
+				Computed:            true, // Terraform requires it to be computed if it's optional.
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.UseStateForUnknown(),
```

Comment:
UseStateForUnknown shouldn't be needed since the default value of an empty list should be used.